go/engine/saltpack_decrypt.go

```diff
@@ -40,21 +43,95 @@ func (e *SaltPackDecrypt) Prereqs() Prereqs {
 
 // RequiredUIs returns the required UIs.
 func (e *SaltPackDecrypt) RequiredUIs() []libkb.UIKind {
-	return nil
+	return []libkb.UIKind{
+		libkb.SaltPackUIKind,
+	}
 }
 
 // SubConsumers returns the other UI consumers for this engine.
 func (e *SaltPackDecrypt) SubConsumers() []libkb.UIConsumer {
-	return []libkb.UIConsumer{}
+	return []libkb.UIConsumer{
+		&Identify2WithUID{},
+	}
+}
+
+func (e *SaltPackDecrypt) lookupSender(ctx *Context, mki *saltpack.MessageKeyInfo, arg *keybase1.SaltPackPromptForDecryptArg) (err error) {
+	defer e.G().Trace("SaltPackDecrypt::lookupSender", func() error { return err })()
+	arg.Username, arg.Uid, err = libkb.KeyLookupByBoxPublicKey(e.G(), mki.SenderKey)
+	return err
+}
+
+func (e *SaltPackDecrypt) identifySender(ctx *Context, arg *keybase1.SaltPackPromptForDecryptArg) (err error) {
+	defer e.G().Trace("SaltPackDecrypt::identifySender", func() error { return err })()
+	iarg := keybase1.Identify2Arg{
+		Uid:                   arg.Uid,
+		UseDelegateUI:         !e.arg.Opts.Interactive,
+		AlwaysBlock:           e.arg.Opts.Interactive,
+		ForceRemoteCheck:      e.arg.Opts.ForceRemoteCheck,
+		NoErrorOnTrackFailure: true,
+		Reason: keybase1.IdentifyReason{
+			Reason: "Identify who encrypted this message",
+			Type:   keybase1.IdentifyReasonType_DECRYPT,
+		},
+	}
+	eng := NewIdentify2WithUID(e.G(), &iarg)
+	if err = eng.Run(ctx); err != nil {
+		return err
+	}
+	switch eng.getTrackType() {
+	case identify2NoTrack:
+		arg.SenderType = keybase1.SaltPackSenderType_NOT_TRACKED
+	case identify2TrackOK:
+		arg.SenderType = keybase1.SaltPackSenderType_TRACKING_OK
+	default:
+		arg.SenderType = keybase1.SaltPackSenderType_TRACKING_BROKE
+	}
+	return nil
+}
```

Comment:
how about the related functions?  Maybe you want everything from `computeSenderArg` on down.